segfaults with swig4.3 and PYTHONFAULTHANDLER=1

`scripts/run-python-tests.sh: line 22:  7262 Segmentation fault: 11  pytest --ignore-glob=*petab* --ignore-glob=*test_splines.py --durations=10 $@`

Happens randomly. So far, twice this week and only on macos. Unclear.

Being updated:

* https://github.com/AMICI-dev/AMICI/actions/runs/11496295512/job/31997628356?pr=2564
* https://github.com/AMICI-dev/AMICI/actions/runs/11483086913/job/31957694802
* https://github.com/AMICI-dev/AMICI/actions/runs/11496540268/job/31998356075?pr=2566
* https://github.com/AMICI-dev/AMICI/actions/runs/11496295512/job/31997628356
* https://github.com/AMICI-dev/AMICI/actions/runs/11496295512/job/31997630402

This happens already during import of the amici-extension. I would currently guess that this is related to some external changes (runner image, brew packages, ...)

https://github.com/dweindl/AMICI/actions/runs/11500013663/job/32009264065:
```
Run scripts/run-python-tests.sh \
Fatal Python error: Segmentation fault

Current thread 0x00000001ef9a0f40 (most recent call first):
  File "<frozen importlib._bootstrap>", line 241 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 1233 in create_module
  File "<frozen importlib._bootstrap>", line 573 in module_from_spec
  File "<frozen importlib._bootstrap>", line 676 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1147 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1176 in _find_and_load
  File "<frozen importlib._bootstrap>", line 241 in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1232 in _handle_fromlist
  File "/Users/runner/work/AMICI/AMICI/python/sdist/amici/amici.py", line 28 in <module>
  File "<frozen importlib._bootstrap>", line 241 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 940 in exec_module
  File "<frozen importlib._bootstrap>", line 690 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1147 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1176 in _find_and_load
  File "<frozen importlib._bootstrap>", line 241 in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1232 in _handle_fromlist
  File "/Users/runner/work/AMICI/AMICI/python/sdist/amici/__init__.py", line 104 in <module>
  File "<frozen importlib._bootstrap>", line 241 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 940 in exec_module
  File "<frozen importlib._bootstrap>", line 690 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1147 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1176 in _find_and_load
  File "/Users/runner/work/AMICI/AMICI/python/tests/conftest.py", line 8 in <module>
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/assertion/rewrite.py", line 184 in exec_module
  File "<frozen importlib._bootstrap>", line 690 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1147 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1176 in _find_and_load
  File "<frozen importlib._bootstrap>", line 1204 in _gcd_import
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/importlib/__init__.py", line 126 in import_module
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/pathlib.py", line 582 in import_path
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 710 in _importconftest
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 659 in _loadconftestmodules
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 619 in _try_load_conftest
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 581 in _set_initial_conftests
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1222 in pytest_load_initial_conftests
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1398 in _preparse
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1494 in parse
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 1140 in pytest_cmdline_parse
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/pluggy/_callers.py", line 103 in _multicall
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/pluggy/_manager.py", line 120 in _hookexec
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 513 in __call__
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 341 in _prepareconfig
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 156 in main
  File "/Users/runner/work/AMICI/AMICI/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 201 in console_main
  File "/Users/runner/work/AMICI/AMICI/venv/bin/pytest", line 8 in <module>

Extension modules: numpy._core._multiarray_umath, numpy.linalg._umath_linalg (total: 2)
scripts/run-python-tests.sh: line 22: 25340 Segmentation fault: 11  pytest -s --ignore-glob=*petab* --ignore-glob=*test_splines.py --durations=10 $@
Error: Process completed with exit code 139.
```

--- a/.github/workflows/test_python_cplusplus.yml
+++ b/.github/workflows/test_python_cplusplus.yml
@@ -288,8 +288,7 @@ jobs:
 
     - name: Python tests
       run: |
-        # ignore warnings until https://github.com/swig/swig/issues/3061 is resolved
-        scripts/run-python-tests.sh -W ignore:: \
+        scripts/run-python-tests.sh \
           test_pregenerated_models.py \
           test_splines_short.py \
           test_misc.py
@@ -351,8 +350,7 @@ jobs:
 
     - name: Python tests
       run: |
-        # ignore warnings until https://github.com/swig/swig/issues/3061 is resolved
-        scripts/run-python-tests.sh -W ignore:: \
+        scripts/run-python-tests.sh \
           --ignore=test_pregenerated_models.py \
           --ignore=test_splines_short.py \
           --ignore=test_misc.py

--- a/python/sdist/amici/__init__.py
+++ b/python/sdist/amici/__init__.py
@@ -16,6 +16,7 @@ from pathlib import Path
 from types import ModuleType
 from typing import Any
 from collections.abc import Callable
+import warnings
 
 
 def _get_amici_path():
@@ -102,7 +103,18 @@ __commit__ = _get_commit_hash()
 # Import SWIG module and swig-dependent submodules if required and available
 if not _imported_from_setup():
     if has_clibs:
-        from . import amici
+        # prevent segfaults under pytest
+        #  see also:
+        #  https://github.com/swig/swig/issues/2881
+        #  https://github.com/AMICI-dev/AMICI/issues/2565
+        with warnings.catch_warnings():
+            warnings.filterwarnings(
+                "ignore",
+                category=DeprecationWarning,
+                message="builtin type .* has no __module__ attribute",
+            )
+            from . import amici
+
         from .amici import *
 
         # has to be done before importing readSolverSettingsFromHDF5

--- a/python/sdist/amici/__init__.py
+++ b/python/sdist/amici/__init__.py
@@ -247,7 +247,17 @@ def import_model_module(
     module_path_matlab = Path(model_root, f"{module_name}.py")
     if not module_path.is_file() and module_path_matlab.is_file():
         with set_path(model_root):
-            return _module_from_path(module_name, module_path_matlab)
+            # prevent segfaults under pytest
+            #  see also:
+            #  https://github.com/swig/swig/issues/2881
+            #  https://github.com/AMICI-dev/AMICI/issues/2565
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    "ignore",
+                    category=DeprecationWarning,
+                    message="builtin type .* has no __module__ attribute",
+                )
+                return _module_from_path(module_name, module_path_matlab)
 
     module = _module_from_path(module_name, module_path)
     module._self = module

--- a/python/sdist/amici/__init__.template.py
+++ b/python/sdist/amici/__init__.template.py
@@ -3,7 +3,7 @@
 import sys
 from pathlib import Path
 import amici
-
+import warnings
 
 # Ensure we are binary-compatible, see #556
 if "TPL_AMICI_VERSION" != amici.__version__:
@@ -16,9 +16,22 @@ if "TPL_AMICI_VERSION" != amici.__version__:
         "version currently installed."
     )
 
-TPL_MODELNAME = amici._module_from_path(
-    "TPL_MODELNAME.TPL_MODELNAME", Path(__file__).parent / "TPL_MODELNAME.py"
-)
+# prevent segfaults under pytest
+#  see also:
+#  https://github.com/swig/swig/issues/2881
+#  https://github.com/AMICI-dev/AMICI/issues/2565
+with warnings.catch_warnings():
+    warnings.filterwarnings(
+        "ignore",
+        category=DeprecationWarning,
+        message="builtin type .* has no __module__ attribute",
+    )
+
+    TPL_MODELNAME = amici._module_from_path(
+        "TPL_MODELNAME.TPL_MODELNAME",
+        Path(__file__).parent / "TPL_MODELNAME.py",
+    )
+
 for var in dir(TPL_MODELNAME):
     if not var.startswith("_"):
         globals()[var] = getattr(TPL_MODELNAME, var)

